### ISSUE TITLE
Radios and Checkboxes fields - Orphaned label & missing fieldset

### ISSUE THREAD

**Issue by ncastle-veic** [solspace/craft3-freeform#2484]
### What happened?

Freeform’s radio group and checkbox group fields appear to be rendered with an invalid labeling pattern for accessibility. When scanning the page with the Wave Accessibility Chrome browser extension, it shows an "Orphaned label" and "Missing fieldset" alert for all radios and checkboxes fields.

The main field label is rendered as a `label` element with a `for` attribute pointing to the base field id, but the actual radio/checkbox inputs are rendered with per-option ids instead. As a result, the visible field label is not associated with any real control, and the group itself does not have a proper accessible name.

One option for fixing this would be to wrap the inputs in a fieldset and add a nested legend element for the label.

![Image](https://github.com/user-attachments/assets/c9ee5f4f-3ff3-4ae8-8f29-bb48fc2f616b)

![Image](https://github.com/user-attachments/assets/45c343c9-eef9-4f18-b0b3-011a258e53fe)

### Errors and Stack Trace (if available)

```shell

```

### How can we reproduce this?

1. Create a new form
2. Add a radios field and/or a checkboxes field
3. Add options
4. Render form on a page
5. Inspect and see that the label incorrectly uses the for attribute and points to the base form handle id.

### Freeform Edition

Pro

### Freeform Version

5.15.3

### Craft Version

5.9.18

### When did this issue start?

- [x] Unsure
- [ ] Fresh install of Freeform
- [ ] After upgrading from older Freeform version
- [ ] After upgrading from older Craft version
- [ ] After a change to hosting configuration
